command/validate.go

```diff
@@ -118,77 +119,34 @@ func (c *ValidateCommand) validate(dir string) tfdiags.Diagnostics {
 func (c *ValidateCommand) showResults(diags tfdiags.Diagnostics, jsonOutput bool) int {
 	switch {
 	case jsonOutput:
-		// FIXME: Eventually we'll probably want to factor this out somewhere
-		// to support machine-readable outputs for other commands too, but for
-		// now it's simplest to do this inline here.
-		type Pos struct {
-			Line   int `json:"line"`
-			Column int `json:"column"`
-			Byte   int `json:"byte"`
-		}
-		type Range struct {
-			Filename string `json:"filename"`
-			Start    Pos    `json:"start"`
-			End      Pos    `json:"end"`
-		}
-		type Diagnostic struct {
-			Severity string `json:"severity,omitempty"`
-			Summary  string `json:"summary,omitempty"`
-			Detail   string `json:"detail,omitempty"`
-			Range    *Range `json:"range,omitempty"`
-		}
 		type Output struct {
```

Comment:
@alisdair I'm not sure whether we anticipate any (significant) changes, but in line with the JSON schema output - Is it worth adding a version to this struct? e.g. `format_version`

Reply:
That's a good question. Now that diagnostics are rendered via this struct, by definition it covers all the current usage of diags, so I don't think it's likely that it will change again.

But I still think it's a good idea to have the version field in place, so I've proposed the change with this PR: https://github.com/hashicorp/terraform/pull/28108

Thanks for the suggestion!